src/components/CategoryVideos/CategoryVideos.tsx

```diff
@@ -0,0 +1,480 @@
+import { add } from 'date-fns'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useParams } from 'react-router'
+import { CSSTransition } from 'react-transition-group'
+
+import { VideoOrderByInput, VideoWhereInput } from '@/api/queries'
+import { languages } from '@/config/languages'
+import { SORT_OPTIONS } from '@/config/sorting'
+import knownLicenses from '@/data/knownLicenses.json'
+import { useMediaMatch } from '@/hooks/useMediaMatch'
+import { Button } from '@/shared/components/Button'
+import { Checkbox } from '@/shared/components/Checkbox'
+import { IconButton } from '@/shared/components/IconButton'
+import { GridItem } from '@/shared/components/LayoutGrid'
+import { PopoverDialog } from '@/shared/components/Popover'
+import { RadioButton } from '@/shared/components/RadioButton'
+import { Select } from '@/shared/components/Select'
+import { Text } from '@/shared/components/Text'
+import { SvgActionFilters, SvgGlyphClose } from '@/shared/icons'
+
+import {
+  Container,
+  ControlsContainer,
+  DateUploadFilterContainer,
+  FiltersContainer,
+  FiltersInnerContainer,
+  MobileFilterContainer,
+  OtherFilterStyledIcon,
+  OtherFilterStyledText,
+  SortContainer,
+} from './CategoryVideos.styles'
+
+import { ActionDialog, ActionDialogProps } from '../ActionDialog'
+import { StyledTitleText } from '../MessageDialog/MessageDialog.style'
+
+type VideoLengthOptions = '0-to-4' | '4-to-10' | '10-to-9999'
+
+export const CategoryVideos = () => {
```

Comment:
The filtering component should be extracted and `CategoryVideos` shouldn't be a part of this PR